src/main.rs

```diff
@@ -12,7 +14,8 @@ use std::path::Path;
 #[derive(Serialize)]
 struct Data {
```

Comment:
Use a more evocative name for the structure instead of struct Data. struct CodeGenData for instance, so that it's clear what it encapsulates.